src/components/modals/LogsModal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react'
+import { Platform, View } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import RNFS from 'react-native-fs'
+import Share, { ShareOptions } from 'react-native-share'
+
+import { getLogOutput } from '../../actions/LogActions'
+import s from '../../locales/strings'
+import { useDispatch } from '../../types/reactRedux'
+import { showError } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { MainButton } from '../themed/MainButton'
+import { ModalCloseArrow, ModalMessage, ModalTitle } from '../themed/ModalParts'
+import { OutlinedTextInput } from '../themed/OutlinedTextInput'
+import { ThemedModal } from '../themed/ThemedModal'
+
+interface LogsModalProps {
+  bridge: AirshipBridge<void>
+}
+
+export const LogsModal = ({ bridge }: LogsModalProps) => {
+  const theme = useTheme()
+  const dispatch = useDispatch()
+  const styles = getStyles(theme)
+  const [spinning, setSpinning] = React.useState(false)
+  const [userMessage, setUserMessage] = React.useState('')
+  const dir = Platform.OS === 'android' ? RNFS.ExternalCachesDirectoryPath : RNFS.DocumentDirectoryPath
+
+  const handleShare = async () => {
+    const logOutput = await dispatch(getLogOutput({ userMessage, send: false }))
+    const logPayload = JSON.stringify(logOutput, null, 2)
+    const logDate = logOutput.isoDate.split('.')[0].split('T')[0]
+
+    const usernameIfLoggedIn = logOutput.loggedInUser != null ? `${logOutput.loggedInUser.userName}` : 'edge'
+    const fn = `${usernameIfLoggedIn}-${logDate}.json`
+    const path = `file://${dir}/${fn}`
+    await RNFS.writeFile(`${dir}/${fn}`, logPayload, 'utf8')
+    const shareOptions: ShareOptions = {
+      title: s.strings.settings_button_export_logs,
+      subject: fn,
+      urls: [path],
+      type: 'application/json'
+    }
+    await Share.open(shareOptions)
+    bridge.resolve()
+  }
+
+  const handleSend = async () => {
+    setSpinning(true)
+    try {
+      await dispatch(getLogOutput({ userMessage, send: true }))
+      bridge.resolve()
+    } catch (e: any) {
+      showError(e.message)
+    }
+    setSpinning(false)
+  }
+
+  const handleChangeText = (text: string) => {
+    setUserMessage(text)
+  }
```

Comment:
You can ditch this callback, and just pass `setUserMessage` straight to the `onChangeText` prop.

Reply:
Done